### PR TITLE
report branch name correctly in s2i builds with complex refs

### PR DESCRIPTION
**Pull request by jim-minter**
https://bugzilla.redhat.com/show_bug.cgi?id=1531565

### CODE REVIEW

File: test/extended/builds/start.go [hunk 1]
```diff
@@ -85,6 +85,10 @@ var _ = g.Describe("[Feature:Builds][Slow] starting a build using CLI", func() {
 					// so simply looking for that string in the mvn output is indicative of being at that level
 					g.By("confirm the correct commit level was retrieved")
 					o.Expect(out).Should(o.ContainSubstring("0.1-SNAPSHOT"))
+
+					istag, err := oc.ImageClient().Image().ImageStreamTags(oc.Namespace()).Get("bc-with-pr-ref:latest", metav1.GetOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+					o.Expect(istag.Image.DockerImageMetadata.Config.Labels).To(o.HaveKeyWithValue("io.openshift.build.commit.ref", "pull/1/head"))
```
Comment (bparees) [inline test/extended/builds/start.go]: might as well check the `OPENSHIFT_BUILD_REFERENCE` env too, right?

File: pkg/build/builder/sti.go [hunk 2]
```diff
@@ -378,6 +378,13 @@ func s2iBuildLabels(build *buildapiv1.Build, sourceInfo *git.SourceInfo) map[str
 	}
 
 	labels = s2iutil.GenerateLabelsFromSourceInfo(labels, toS2ISourceInfo(sourceInfo), builderutil.DefaultDockerLabelNamespace)
+	if build != nil && build.Spec.Source.Git != nil && build.Spec.Source.Git.Ref != "" {
+		// override the io.openshift.build.commit.ref label to match what we
+		// were originally told to check out, as well as the
+		// OPENSHIFT_BUILD_REFERENCE environment variable.  This can sometimes
+		// differ from git's view (see PotentialPRRetryAsFetch for details).
+		labels[builderutil.DefaultDockerLabelNamespace+"build.commit.ref"] = build.Spec.Source.Git.Ref
```
Comment (bparees) [inline pkg/build/builder/sti.go]: i'm guessing docker.go also has similar logic that needs to be updated?

Comment (jim-minter) [reply pkg/build/builder/sti.go]: updated, thanks

**Pull request merged**